### PR TITLE
DEXA-281 Add VideoView

### PR DESCRIPTION
**Pull request by krzysztof-drobek**

### CODE REVIEW

File: android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    }
+  }
+
+  private fun findMediaStream(peerId: String, streamId: String): MediaStream? =
+    conferenceService
+      .findParticipantById(peerId)
+      ?.streams()
+      ?.find { it.peerId() == peerId && it.label() == streamId }
+
+  private enum class VideoViewCommands(val commandId: Int, val commandName: String) {
```
Comment (romansavka-mq) [inline android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt:99]: I think it should be singular VideoCommand

File: android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      } ?: videoView.unAttach()
+      false
+    } catch (exception: Exception) {
+      false
+    }
+  }
+
+  private fun findMediaStream(peerId: String, streamId: String): MediaStream? =
+    conferenceService
+      .findParticipantById(peerId)
```
Comment (romansavka-mq) [inline android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt:95]: Is it correct that peerId is used to find participant?

File: android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  private fun detach(videoView: VideoView) = try {
+    videoView.unAttach()
+    true
+  } catch (exception: Exception) {
+    false
+  }
+
+  private fun attach(peerId: String?, streamId: String?, videoView: VideoView): Boolean {
+    return try {
```
Comment (romansavka-mq) [inline android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt:81]: We reduce returns count if place unAttach to `run` block
``` kotlin
return try {
      if (peerId == null || streamId == null) return false
      findMediaStream(peerId, streamId)?.let {
        videoView.attach(peerId, it)
        true
      } ?: run {
        videoView.unAttach()
        false
      }
    } catch (exception: Exception) {
      false
    }
```

File: android/src/main/java/io/dolby/sdk/reactnative/eventemitters/RNVideoViewEventEmitter.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        putBoolean(KEY_RESULT, result)
+      }
+      .also { send(VideoViewEvent.CommandCallback.withData(it)) }
+  }
+
+  /**
+   * Video View events
+   */
+  private object VideoViewEvent {
+    object CommandCallback : RNEvent("DIOVideoView_Event")
```
Comment (romansavka-mq) [inline android/src/main/java/io/dolby/sdk/reactnative/eventemitters/RNVideoViewEventEmitter.kt:31]: For consistency it should be uppercase with underscores

File: android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt (hunk 2 above)
Comment (krzysztof-drobek) [reply android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt:95]: It was based on `MediaStream` `peerId` but I agree that `participantId` is more suitable in this case.

File: android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt (hunk 3 above)
Comment (krzysztof-drobek) [reply android/src/main/java/io/dolby/sdk/reactnative/view/VideoViewManager.kt:79]: Changed

**Review by romansavka-mq — APPROVED** [review]

**Review by aamq — APPROVED** [review]

**Pull request merged**